### PR TITLE
Return a JSON formatted string when a message is requested

### PR DESCRIPTION
**Pull request by rysimone**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -19,14 +19,36 @@
...
+    json += "\"tide\": ";
+    json += "\"" + surfReport.get(0) + "\"";
+    json += ", ";
+    json += "\"waterTemp\": ";
+    json += "\"" + surfReport.get(1) + "\"";
+    json += ", ";
+    json += "\"surfHeight\": ";
+    json += "\"" + surfReport.get(2) + "\"";
+    json += "}";
+    return json;
```
Comment (megankj) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:52]: Instead of building the JSON string yourself (which could be error prone), I suggest using Gson to convert this to json.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -19,14 +19,36 @@
...
+
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  private String convertToJson(){
+    ArrayList<String> surfReport = new ArrayList<String>(3);
+    surfReport.add("low");
+    surfReport.add("60");
+    surfReport.add("3ft");
```
Comment (megankj) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:41]: This method is actually doing two things:
1) creating an array of hardcoded strings
2) returning those strings as json

I recommend converting this method to taking an ArrayList<String> as an argument and returning the JSON string, since that's more extensible.  This hard-coded string can be created either during construction or in the doGet method.   

Furthermore, I think we can find a better way to store this data that better captures the meaning of these values.  Perhaps it can be represented as an ArrayList<Pair<String, String>> since you're data looks like {(tide, low), (waterTem, 60), (surfHeight, 3ft)}.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -30,5 +30,6 @@ function addRandomFact() {
 async function getMessageFromServer() {
   const response = await fetch('/data');
   const quote = await response.text();
+  console.log(quote);
```
Comment (megankj) [inline portfolio/src/main/webapp/script.js:33]: console.log is good for testing, but shouldn't be included in submitted code :)

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -30,5 +30,6 @@ function addRandomFact() {
...
   document.getElementById('message-container').innerHTML = quote;
```
Comment (megankj) [inline portfolio/src/main/webapp/script.js:34]: So now that the response comes back as JSON, you can parse the objects in the response here to show a more formatted response.  Are you planning on doing that in a follow up?

Comment (rysimone) [reply portfolio/src/main/webapp/script.js:34]: Yes I am doing that in the next task.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (rysimone) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:41]: Do you know a good import statement for the Pair class? I tried using javafx.util.Pair but when it compiles I'm told that it doesn't exist.

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (rysimone) [reply portfolio/src/main/webapp/script.js:33]: Ok thanks that's good to know.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (megankj) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:41]: I think you'd need to add the dependency in pom.xml like this:
https://mvnrepository.com/artifact/org.openjfx/javafx/11

Similar to how you added a dependency for gson.

Alternatively, maybe a HashMap makes more sense in this case since in essence you're storing key/value pairs.

File: portfolio/pom.xml [hunk 5]
```diff
@@ -23,6 +23,18 @@
       <version>4.0.1</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>com.google.code.gson</groupId>
+      <artifactId>gson</artifactId>
+      <version>2.8.6</version>
+    </dependency>
+    <!-- https://mvnrepository.com/artifact/org.openjfx/javafx -->
+    <dependency>
```
Comment (tjgoog) [inline portfolio/pom.xml:32]: No longer needed since you're using HashMap directly (which is good IMO)

File: walkthroughs/week-3-server/examples/random-quotes/src/main/webapp/index.html [hunk 6]
```diff
@@ -10,7 +10,7 @@
     <div id="content">
       <h1>Random Quotes</h1>
       <p>Click here to see a random quote:</p>
-      <button onclick="getRandomQuote()">Fetch a quote!</button>
```
Comment (tjgoog) [inline walkthroughs/week-3-server/examples/random-quotes/src/main/webapp/index.html:13]: The changes to this and the script.js file seem unrelated and should probably not be in this PR, though maybe github is confusing me again somehow.

Comment (rysimone) [reply walkthroughs/week-3-server/examples/random-quotes/src/main/webapp/index.html:13]: That was a change I made during last week's project. I am confused as well as why that is shown as a change in this PR.

Comment (tjgoog) [reply walkthroughs/week-3-server/examples/random-quotes/src/main/webapp/index.html:13]: Could you revert these changes if easily possible? If not, it's fine. I _think_ it should be as simple as doing a new commit that changes the names back.

File: walkthroughs/week-3-server/examples/random-quotes/src/main/webapp/script.js [hunk 7]
```diff
@@ -56,6 +56,7 @@ function addQuoteToDom(quote) {
  * whichever syntax makes the most sense to you.
  */
 function getRandomQuoteUsingArrowFunctions() {
+  console.log('Fetching quote with arrow functions.')
```
Comment (megankj) [inline walkthroughs/week-3-server/examples/random-quotes/src/main/webapp/script.js:59]: are these changes intended to go in this PR?

File: walkthroughs/week-3-server/examples/random-quotes/src/main/webapp/index.html (hunk 6 above)
Comment (megankj) [reply walkthroughs/week-3-server/examples/random-quotes/src/main/webapp/index.html:13]: I'm a little confused - the java code you're adding in this CL is unrelated to the HTML/JS code, and furthermore it doesn't look like anything in the HTML/JS is even fetching from /data.

File: walkthroughs/week-3-server/examples/random-quotes/src/main/webapp/script.js (hunk 7 above)
Comment (rysimone) [reply walkthroughs/week-3-server/examples/random-quotes/src/main/webapp/script.js:59]: No I am confused as to why these are showing up in this PR.

Comment (rysimone) [reply walkthroughs/week-3-server/examples/random-quotes/src/main/webapp/script.js:59]: These changes you guys are commenting on are from last week's project where I needed to change the Fetch a Quote button to a Get a Random Quote fact. These changes are not in any file on my branch so i'm not sure why these would be included.

Comment (megankj) [reply walkthroughs/week-3-server/examples/random-quotes/src/main/webapp/script.js:59]: do you need to re-commit?
